pkg/sql/logictest/tests/fakedist/fakedist_test: TestLogic_optimizer_timeout failed

pkg/sql/logictest/tests/fakedist/fakedist_test.TestLogic_optimizer_timeout [failed](https://mesolite.cluster.engflow.com/invocations/default/b17bfeb8-7d65-4339-ae83-5693fe0eab44?testReportRun=3&testReportShard=34&testReportAttempt=1#targets-Ly9wa2cvc3FsL2xvZ2ljdGVzdC90ZXN0cy9mYWtlZGlzdDpmYWtlZGlzdF90ZXN0) on release-25.2.12-rc @ [7fc1e95228ad6610c2ea21d8ba55d0d6513c319f](https://github.com/cockroachdb/cockroach/commits/7fc1e95228ad6610c2ea21d8ba55d0d6513c319f):


```
=== RUN   TestLogic_optimizer_timeout
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLogic_optimizer_timeout2945103546
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:141: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
[11:13:37] --- done: /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist/fakedist_test_/fakedist_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/optimizer_timeout with config fakedist: 3 tests, 0 failures
    logic.go:4326: 
        /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist/fakedist_test_/fakedist_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/optimizer_timeout:55: error while processing
    logic.go:4326: 
        /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist/fakedist_test_/fakedist_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/optimizer_timeout:55: 
        expected success, but found
        (57014) query execution canceled due to statement timeout
        errors.go:553: in init()
    panic.go:635: -- test log scope end --
test logs left over in: outputs.zip/logTestLogic_optimizer_timeout2945103546
--- FAIL: TestLogic_optimizer_timeout (7.53s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=34</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_optimizer_timeout.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-59256

--- a/pkg/sql/logictest/testdata/logic_test/optimizer_timeout
+++ b/pkg/sql/logictest/testdata/logic_test/optimizer_timeout
@@ -1,4 +1,7 @@
 # LogicTest: !local-prepared
+
+skip under race
+
 statement ok
 CREATE TABLE table1
 (

